server/container/guice/guice-common/src/test/java/org/apache/james/PeriodicalHealthChecksTest.java

```diff
@@ -60,46 +89,96 @@ void setUp() {
         scheduler = VirtualTimeScheduler.getOrSet();
         testee = new PeriodicalHealthChecks(ImmutableSet.of(mockHealthCheck1, mockHealthCheck2),
             scheduler,
-            new PeriodicalHealthChecksConfiguration(Duration.ofSeconds(PERIOD)));
+            new PeriodicalHealthChecksConfiguration(PERIOD));
     }
 
     @AfterEach
     void tearDown() {
         testee.stop();
     }
-    
+
     @Test
     void startShouldCallHealthCheckAtLeastOnce() {
         testee.start();
 
-        scheduler.advanceTimeBy(Duration.ofSeconds(PERIOD));
+        scheduler.advanceTimeBy(PERIOD);
         verify(mockHealthCheck1, atLeast(1)).check();
     }
 
+    @Test
+    void startShouldLogPeriodicallyWhenUnhealthy() {
+        ListAppender<ILoggingEvent> loggingEvents = getListAppenderForClass(PeriodicalHealthChecks.class);
+
+        TestingHealthCheck unhealthy = () -> Mono.just(Result.unhealthy(TestingHealthCheck.COMPONENT_NAME, "cause"));
+        testee = new PeriodicalHealthChecks(ImmutableSet.of(unhealthy),
+            scheduler,
+            new PeriodicalHealthChecksConfiguration(PERIOD));
+        testee.start();
+
+        scheduler.advanceTimeBy(PERIOD);
+        assertThat(loggingEvents.list).hasSize(1)
+            .allSatisfy(loggingEvent -> {
+                assertThat(loggingEvent.getLevel()).isEqualTo(Level.ERROR);
+                assertThat(loggingEvent.getFormattedMessage()).isEqualTo("UNHEALTHY: testing : Optional[cause]");
```

Comment:
are we really outputing `Optional[cause]`?
also we should match only some substrings of the message